packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -121,20 +117,76 @@ function tryGetNodeProvider(): NodeProvider | undefined {
   }
 }
 
-export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => {
+export const AlephiumConnectProvider: React.FC<{
+  network: NetworkId
+  addressGroup?: number
+  keyType?: KeyType
+  children?: React.ReactNode
+}> = ({ network, addressGroup, keyType, children }) => {
   // Only allow for mounting AlephiumConnectProvider once, so we avoid weird global
   // state collisions.
   const context = useContext(AlephiumConnectContext)
   if (context) {
     throw new Error('Multiple, nested usages of AlephiumConnectProvider detected. Please use only one.')
   }
 
-  const [account, setAccount] = useState<Account & { network: NetworkId }>()
+  const lastConnectedAccount = useMemo(() => {
+    const result = getLastConnectedAccount()
+    if (result === undefined) {
+      return undefined
+    }
+    if (
+      result.network === network &&
+      (addressGroup === undefined || result.account.group === addressGroup) &&
+      (keyType === undefined || result.account.keyType === keyType)
+    ) {
```

Comment:
We need to check if the `addressGroup` or `network` has changed, so I moved them to the `AlephiumConnectContext`.

Reply:
Makes more sense to be in `AlephiumConnectContext`